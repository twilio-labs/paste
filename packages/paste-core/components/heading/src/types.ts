export type AsTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'label' | 'span' | 'header';
export type HeadingVariants = 'heading10' | 'heading20' | 'heading30' | 'heading40' | 'heading50' | 'heading60';

export interface HeadingProps extends React.HTMLAttributes<HTMLHeadingElement> {
  as: AsTags;
  className?: never;
  id?: string;
  marginBottom?: 'space0';
  variant: HeadingVariants;
}
