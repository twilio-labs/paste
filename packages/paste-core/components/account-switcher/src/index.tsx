export * from "./AccountSwitcher";
export * from "./AccountSwitcherGroup";
export * from "./AccountSwitcherSeparator";
export * from "./AccountSwitcherItem";
export * from "./AccountSwitcherItemRadio";
export * from "./AccountSwitcherBadge";
export * from "./useAccountSwitcher";
