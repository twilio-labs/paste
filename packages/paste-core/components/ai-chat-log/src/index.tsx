export { AIChatMessage } from "./AIChatMessage";
export type { AIChatMessageProps } from "./AIChatMessage";
export { AIChatMessageActionGroup } from "./AIChatMessageActionGroup";
export type { AIChatMessageActionGroupProps } from "./AIChatMessageActionGroup";
export { AIChatMessageBody } from "./AIChatMessageBody";
export type { AIChatMessageBodyProps } from "./AIChatMessageBody";
export { AIChatMessageAuthor } from "./AIChatMessageAuthor";
export type { AIChatMessageAuthorProps } from "./AIChatMessageAuthor";
export { AIChatMessageActionCard } from "./AIChatMessageActionCard";
export type { AIChatMessageActionCardProps } from "./AIChatMessageActionCard";
export { AIChatMessageLoading } from "./AIChatMessageLoading";
export type { AIChatMessageLoadingProps } from "./AIChatMessageLoading";

export { AIChatLog } from "./AIChatLog";
export type { AIChatLogProps } from "./AIChatLog";
export { useAIChatLogger } from "./useAIChatLogger";
export type { UseAIChatLogger, AIPartialIDChat, AIChat } from "./useAIChatLogger";
export { AIChatLogger } from "./AIChatLogger";
export type { AIChatLoggerProps } from "./AIChatLogger";

export type { AIMessageVariants } from "./AIMessageContext";
