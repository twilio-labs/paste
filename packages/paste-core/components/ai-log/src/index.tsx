export { AIChatMessage } from "./AIChatMessage";
export type { AIChatMessageProps } from "./AIChatMessage";
export { AIChatMessageMeta } from "./AIChatMessageMeta";
export type { AIChatMessageMetaProps } from "./AIChatMessageMeta";
export { AIChatMessageMetaItem } from "./AIChatMessageMetaItem";
export type { AIChatMessageMetaItemProps } from "./AIChatMessageMetaItem";
export { AIChatMessageContent } from "./AIChatMessageContent";
export type { AIChatMessageContentProps } from "./AIChatMessageContent";

export { AIChatLog } from "./AIChatLog";
export type { AIChatLogProps } from "./AIChatLog";
export { useAIChatLogger } from "./useAIChatLogger";
export type { UseAIChatLogger } from "./useAIChatLogger";
export { AIChatLogger } from "./AIChatLogger";
export type { AIChatLoggerProps } from "./AIChatLogger";
