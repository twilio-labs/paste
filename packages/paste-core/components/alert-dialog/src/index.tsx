import * as React from 'react';
import * as PropTypes from 'prop-types';
import {useUID} from '@twilio-paste/uid-library';
import {useTransition} from '@twilio-paste/animation-library';
import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
import type {BoxProps} from '@twilio-paste/box';
import {ModalDialogOverlay} from '@twilio-paste/modal';
import {AlertDialogHeader} from './AlertDialogHeader';
import {AlertDialogBody} from './AlertDialogBody';
import {AlertDialogContent} from './AlertDialogContent';
import {AlertDialogFooter} from './AlertDialogFooter';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getAnimationStates = (): any => ({
  from: {opacity: 0, transform: `scale(0.675)`},
  enter: {opacity: 1, transform: `scale(1)`},
  leave: {opacity: 0, transform: `scale(0.675)`},
  // https://www.react-spring.io/docs/hooks/api
  config: {
    mass: 0.5,
    tension: 370,
    friction: 26,
  },
});

export interface AlertDialogProps extends React.HTMLAttributes<HTMLElement>, Pick<BoxProps, 'element'> {
  children: NonNullable<React.ReactNode>;
  destructive?: boolean;
  heading: string;
  isOpen: boolean;
  onConfirm: () => void;
  onConfirmLabel: string;
  onDismiss: () => void;
  onDismissLabel: string;
}

export const AlertDialog = React.forwardRef<HTMLDivElement, AlertDialogProps>(
  (
    {
      children,
      destructive,
      element = 'ALERT_DIALOG',
      heading,
      isOpen,
      onConfirm,
      onConfirmLabel,
      onDismiss,
      onDismissLabel,
      ...props
    },
    ref
  ) => {
    const transitions = useTransition(isOpen, getAnimationStates());
    const headingID = useUID();
    const bodyID = useUID();

    return (
      <>
        {transitions(
          (styles, item) =>
            item && (
              <ModalDialogOverlay isOpen={isOpen} style={{opacity: styles.opacity}}>
                <Box
                  // @ts-expect-error Render overlay as box for customization
                  as={AlertDialogContent}
                  {...safelySpreadBoxProps(props)}
                  aria-labelledby={headingID}
                  aria-describedby={bodyID}
                  element={element}
                  ref={ref}
                  role="alertdialog"
                  style={styles}
                >
                  <AlertDialogHeader headingID={headingID} element={`${element}_HEADER`}>
                    {heading}
                  </AlertDialogHeader>
                  <AlertDialogBody bodyID={bodyID} element={`${element}_BODY`}>
                    {children}
                  </AlertDialogBody>
                  <AlertDialogFooter
                    destructive={destructive}
                    element={`${element}_FOOTER`}
                    onDismiss={onDismiss}
                    onDismissLabel={onDismissLabel}
                    onConfirm={onConfirm}
                    onConfirmLabel={onConfirmLabel}
                  />
                </Box>
              </ModalDialogOverlay>
            )
        )}
      </>
    );
  }
);

AlertDialog.displayName = 'AlertDialog';

AlertDialog.propTypes = {
  children: PropTypes.node.isRequired,
  destructive: PropTypes.bool,
  element: PropTypes.string,
  heading: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onConfirmLabel: PropTypes.string.isRequired,
  onDismiss: PropTypes.func.isRequired,
  onDismissLabel: PropTypes.string.isRequired,
};
