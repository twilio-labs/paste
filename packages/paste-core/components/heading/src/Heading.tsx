import type { TextStyleProps } from "@twilio-paste/text";
import { Text, safelySpreadTextProps } from "@twilio-paste/text";
import * as React from "react";

import type { HeadingProps, HeadingVariants } from "./types";

function getHeadingProps(headingVariant?: HeadingVariants, marginBottom?: "space0"): TextStyleProps {
  switch (headingVariant) {
    case "heading10":
      return {
        marginBottom: marginBottom || "space70",
        fontSize: "fontSize90",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight90",
        letterSpacing: "-.02em",
      };
    case "heading30":
      return {
        marginBottom: marginBottom || "space50",
        fontSize: "fontSize60",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight60",
        letterSpacing: "-.02em",
      };
    case "heading40":
      return {
        marginBottom: marginBottom || "space30",
        fontSize: "fontSize40",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight40",
        letterSpacing: "-.02em",
      };
    case "heading50":
      return {
        marginBottom: marginBottom || "space30",
        fontSize: "fontSize30",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight30",
        letterSpacing: "-.02em",
      };
    case "heading60":
      return {
        marginBottom: marginBottom || "space30",
        fontSize: "fontSize20",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight20",
        letterSpacing: "-.02em",
      };
    /**
     * heading20 is out of order because its also default.
     * Default is at the bottom of switch statement for readability.
     */
    case "heading20":
    default:
      return {
        marginBottom: marginBottom || "space60",
        fontSize: "fontSize70",
        fontWeight: "fontWeightSemibold",
        lineHeight: "lineHeight70",
        letterSpacing: "-.02em",
      };
  }
}

const Heading = React.forwardRef<HTMLHeadingElement, HeadingProps>(
  ({ as, children, display = "block", element = "HEADING", id, marginBottom, variant, ...props }, ref) => {
    return (
      <Text
        {...safelySpreadTextProps(props)}
        {...getHeadingProps(variant, marginBottom)}
        as={as}
        display={display}
        element={element}
        id={id}
        color="colorText"
        variant={variant}
        ref={ref}
      >
        {children}
      </Text>
    );
  },
);

Heading.displayName = "Heading";
export { Heading };
