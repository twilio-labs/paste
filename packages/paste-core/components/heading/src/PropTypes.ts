import * as PropTypes from 'prop-types';
import {HeadingVariants, AsTags} from './types';

export const HeadingPropTypes = {
  as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'div', 'label', 'span'] as AsTags[]).isRequired,
  marginBottom: PropTypes.oneOf(['space0']) as PropTypes.Validator<'space0' | null | undefined> | undefined,
  variant: PropTypes.oneOf([
    'heading10',
    'heading20',
    'heading30',
    'heading40',
    'heading50',
    'heading60',
  ] as HeadingVariants[]).isRequired,
};
