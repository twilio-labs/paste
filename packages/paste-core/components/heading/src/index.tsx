import * as React from 'react';
import * as PropTypes from 'prop-types';
import {Text} from '@twilio-paste/text';

export type asTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'label' | 'span' | 'header';
export type HeadingVariants = 'heading10' | 'heading20' | 'heading30' | 'heading40' | 'heading50' | 'heading60';

export interface HeadingProps {
  as: asTags;
  id?: string;
  className?: never;
  marginBottom?: 'space0';
  variant?: HeadingVariants;
}

function getHeadingProps(headingVariant?: HeadingVariants, marginBottom?: 'space0'): {} {
  switch (headingVariant) {
    case 'heading10':
      return {
        marginBottom: marginBottom || 'space70',
        fontSize: 'fontSize90',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight90',
      };
    case 'heading30':
      return {
        marginBottom: marginBottom || 'space50',
        fontSize: 'fontSize60',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight60',
      };
    case 'heading40':
      return {
        marginBottom: marginBottom || 'space40',
        fontSize: 'fontSize40',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight40',
      };
    case 'heading50':
      return {
        marginBottom: marginBottom || 'space30',
        fontSize: 'fontSize30',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight30',
      };
    case 'heading60':
      return {
        marginBottom: marginBottom || 'space30',
        fontSize: 'fontSize20',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight20',
      };
    /**
     * heading20 is out of order because its also default.
     * Default is at the bottom of switch statement for readability.
     */
    case 'heading20':
    default:
      return {
        marginBottom: marginBottom || 'space60',
        fontSize: 'fontSize70',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight70',
      };
  }
}

const Heading: React.FC<HeadingProps> = ({as, children, id, marginBottom, variant}) => {
  return (
    <Text {...getHeadingProps(variant, marginBottom)} as={as} display="block" id={id} color="colorText">
      {children}
    </Text>
  );
};
Heading.displayName = 'Heading';

Heading.defaultProps = {
  variant: 'heading20',
};

if (process.env.NODE_ENV === 'development') {
  Heading.propTypes = {
    as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'div', 'label', 'span'] as asTags[]).isRequired,
    marginBottom: PropTypes.oneOf(['space0']),
    variant: PropTypes.oneOf(['heading10', 'heading20', 'heading30', 'heading40', 'heading50', 'heading60']),
  };
}

export {Heading};
