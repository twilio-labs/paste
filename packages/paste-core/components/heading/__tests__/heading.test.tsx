import * as React from 'react';
import renderer from 'react-test-renderer';
import {Theme} from '@twilio-paste/theme';
import {Heading} from '../src';

describe('Heading', () => {
  it('it should render an H1 at fontSize90', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h1" variant="heading10">
            This is an H1
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an H2 at fontSize70', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h2" variant="heading20">
            This is an H2
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an H3 at fontSize60', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h3" variant="heading30">
            This is an H3
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an H4 at fontSize40', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h4" variant="heading40">
            This is an H4
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an H5 at fontSize30', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h5" variant="heading50">
            This is an H5
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an H6 at fontSize20', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h6" variant="heading60">
            This is an H6
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render an italic H2 at fontSize50', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Heading as="h2" variant="heading20">
            <i>This is an italic H2</i>
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
