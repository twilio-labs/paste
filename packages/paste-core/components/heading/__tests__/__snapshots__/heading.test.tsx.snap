// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading it should render an H1 at fontSize90 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.5rem;
  font-size: 2rem;
  font-weight: 500;
  line-height: 2.5rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h1
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize90"
    fontWeight="fontWeightSemibold"
  >
    This is an H1
  </h1>
</div>
`;

exports[`Heading it should render an H2 at fontSize70 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.25rem;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 2rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h2
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize70"
    fontWeight="fontWeightSemibold"
  >
    This is an H2
  </h2>
</div>
`;

exports[`Heading it should render an H3 at fontSize60 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1rem;
  font-size: 1.25rem;
  font-weight: 500;
  line-height: 1.75rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h3
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize60"
    fontWeight="fontWeightSemibold"
  >
    This is an H3
  </h3>
</div>
`;

exports[`Heading it should render an H4 at fontSize40 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 0.75rem;
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.75rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h4
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize40"
    fontWeight="fontWeightSemibold"
  >
    This is an H4
  </h4>
</div>
`;

exports[`Heading it should render an H5 at fontSize30 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 0.5rem;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.5rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h5
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize30"
    fontWeight="fontWeightSemibold"
  >
    This is an H5
  </h5>
</div>
`;

exports[`Heading it should render an H6 at fontSize20 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.25rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h6
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize20"
    fontWeight="fontWeightSemibold"
  >
    This is an H6
  </h6>
</div>
`;

exports[`Heading it should render an italic H2 at fontSize50 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 1.25rem;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 2rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-1"
>
  <h2
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize70"
    fontWeight="fontWeightSemibold"
  >
    <i>
      This is an italic H2
    </i>
  </h2>
</div>
`;

exports[`Heading it should render with no margin 1`] = `
.emotion-6 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-6 *,
.emotion-6 *::after,
.emotion-6 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 2rem;
  font-weight: 500;
  line-height: 2.5rem;
  display: block;
  color: #282a2b;
}

.emotion-1 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 2rem;
  display: block;
  color: #282a2b;
}

.emotion-2 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 1.25rem;
  font-weight: 500;
  line-height: 1.75rem;
  display: block;
  color: #282a2b;
}

.emotion-3 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.75rem;
  display: block;
  color: #282a2b;
}

.emotion-4 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.5rem;
  display: block;
  color: #282a2b;
}

.emotion-5 {
  margin: 0;
  padding: 0;
  margin-bottom: 0;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.25rem;
  display: block;
  color: #282a2b;
}

<div
  className="emotion-6"
>
  <h2
    className="emotion-0"
    color="colorText"
    display="block"
    fontSize="fontSize90"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
  <h2
    className="emotion-1"
    color="colorText"
    display="block"
    fontSize="fontSize70"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
  <h2
    className="emotion-2"
    color="colorText"
    display="block"
    fontSize="fontSize60"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
  <h2
    className="emotion-3"
    color="colorText"
    display="block"
    fontSize="fontSize40"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
  <h2
    className="emotion-4"
    color="colorText"
    display="block"
    fontSize="fontSize30"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
  <h2
    className="emotion-5"
    color="colorText"
    display="block"
    fontSize="fontSize20"
    fontWeight="fontWeightSemibold"
  >
    no margin heading
  </h2>
</div>
`;
